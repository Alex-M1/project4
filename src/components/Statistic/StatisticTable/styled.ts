import styled from 'styled-components';
import { IStyled } from 'common/types/styledTypes';
import { changeTheme } from 'src/helpers/changeTheme';

export const StWrapper = styled.div <IStyled>`
  width: 40%;
  height: 75%;
  border-radius: 12px;
  overflow: hidden;
  color: ${(props) => changeTheme(props, 'textColor')};
  background: ${(props) => changeTheme(props, 'backgroundSecondary')};
  box-shadow: 2px 2px 5px ${(props) => changeTheme(props, 'gray')};
`;
export const StHeaderWrapper = styled.div <IStyled>`
  width: 100%;
  height: 40px;
  margin: 10px 0;
  
  & h1 {
    padding: 5px 0;
    text-align: center;
  }

  & div {
    display: flex;
    justify-content: space-around;
    width: 85%;

    height: 30px;
    margin: 10px;

    & span:first-child {
      text-align: center;
    }
  }
`;
export const StResultsWrapper = styled.div <IStyled>`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  max-height: 455px;
  margin-top: 40px;
  overflow: scroll;

  & div {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    width: 90%;
    min-height: 20px;
    height: auto;
    margin: 5px 0;
    padding: 15px 0;
    border-bottom: 1px solid ${(props) => changeTheme(props, 'gray')};
    & span {
      text-align: center;
    }

    & span:first-child {
      width: 45%;
    }
  }
`;
