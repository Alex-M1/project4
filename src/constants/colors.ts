export const colors = {
  light: {
    background: '#fff',
    backgroundSecondary: '#ECECEC',
    headerBg: '#2B5278',
    backgroundItem: '#d7d8e2',
    button: '#2B5278',
    buttonColor: '#fff',
    buttonHover: '#346392',
    input: '#D1D1D1',
    textColor: '#000',
    textColorHeader: '#fff',
    placeholderColor: '#55667B',
    AddRoomBtn: '#fff',
    AddRoomBtnHover: '#fff',
    backgroundModal: 'rgba(0,0,0, 0.8)',
    backgroundModalShadow: 'rgba(0,0,0, 0.8)',
    backgroundModalContainer: '#dcd9d9',
    selectBackground: '#d4d4d4',
    iconTheme: '#ff0',
    iconThemeMoon: 'rgba(0,0,0,0)',
    iconThemeSun: '#ff0',
    ticTacColor: '#000',
    whiteChecker: '#fff',
    blackChecker: '#777',
    gray: '#ccc',
    white: '#fff',
    black: '#000',
    blackCellBg: '#777',
    hoverLight: '#ddd',
    gold: '#FFD700',
  },
  dark: {
    background: '#0E1621',
    backgroundSecondary: '#17212B',
    headerBg: '#2B5278',
    backgroundItem: '#25323f',
    button: '#2B5278',
    buttonColor: '#fff',
    buttonHover: '#346392',
    input: '#242F3D',
    textColor: '#d4d4d4',
    textColorHeader: '#d4d4d4',
    placeholderColor: '#55667B',
    AddRoomBtn: '#fff',
    AddRoomBtnHover: '#fff',
    backgroundModal: 'rgba(0,0,0, 0.8)',
    backgroundModalShadow: '#757575',
    backgroundModalContainer: '#17212B',
    selectBackground: '#d4d4d4',
    iconTheme: '#fff',
    iconThemeMoon: '#2B5278',
    iconThemeSun: 'rgba(0,0,0,0)',
    ticTacColor: '#fff',
    whiteChecker: '#fff',
    blackChecker: '#777',
    gray: '#ccc',
    white: '#fff',
    black: '#000',
    blackCellBg: '#777',
    hoverLight: '#ddd',
    gold: '#FFD700',
  },
};
