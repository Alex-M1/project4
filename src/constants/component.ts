export const CELL = [
  {
    id: 1,
    color: 'black',
  },
  {
    id: 2,
    color: 'white',
  },
  {
    id: 3,
    color: 'black',
  },
  {
    id: 4,
    color: 'white',
  },
  {
    id: 5,
    color: 'black',
  },
  {
    id: 6,
    color: 'white',
  },
  {
    id: 7,
    color: 'black',
  },
  {
    id: 8,
    color: 'white',
  },
  {
    id: 9,
    color: 'black',
  },
  {
    id: 10,
    color: 'white',
  },
  {
    id: 11,
    color: 'black',
  },
  {
    id: 12,
    color: 'white',
  },
  {
    id: 13,
    color: 'black',
  },
  {
    id: 14,
    color: 'white',
  },
  {
    id: 15,
    color: 'black',
  },
  {
    id: 16,
    color: 'white',
  },
  {
    id: 17,
    color: 'black',
  },
  {
    id: 18,
    color: 'white',
  },
  {
    id: 19,
    color: 'black',
  },
  {
    id: 20,
    color: 'white',
  },
  {
    id: 21,
    color: 'black',
  },
  {
    id: 22,
    color: 'white',
  },
  {
    id: 23,
    color: 'black',
  },
  {
    id: 24,
    color: 'white',
  },
  {
    id: 25,
    color: 'black',
  },
  {
    id: 26,
    color: 'white',
  },
  {
    id: 27,
    color: 'black',
  },
  {
    id: 28,
    color: 'white',
  },
  {
    id: 29,
    color: 'black',
  },
  {
    id: 30,
    color: 'white',
  },
  {
    id: 31,
    color: 'black',
  },
  {
    id: 32,
    color: 'white',
  },
  {
    id: 33,
    color: 'black',
  },
  {
    id: 34,
    color: 'white',
  },
  {
    id: 35,
    color: 'black',
  },
  {
    id: 36,
    color: 'white',
  },
  {
    id: 37,
    color: 'black',
  },
  {
    id: 38,
    color: 'white',
  },
  {
    id: 39,
    color: 'black',
  },
  {
    id: 40,
    color: 'white',
  },
  {
    id: 41,
    color: 'black',
  },
  {
    id: 42,
    color: 'white',
  },
  {
    id: 43,
    color: 'black',
  },
  {
    id: 44,
    color: 'white',
  },
  {
    id: 45,
    color: 'black',
  },
  {
    id: 46,
    color: 'white',
  },
  {
    id: 47,
    color: 'black',
  },
  {
    id: 48,
    color: 'white',
  },
  {
    id: 49,
    color: 'black',
  },
  {
    id: 50,
    color: 'white',
  },
  {
    id: 51,
    color: 'black',
  },
  {
    id: 52,
    color: 'white',
  },
  {
    id: 53,
    color: 'black',
  },
  {
    id: 54,
    color: 'white',
  },
  {
    id: 55,
    color: 'black',
  },
  {
    id: 56,
    color: 'white',
  },
  {
    id: 57,
    color: 'black',
  },
  {
    id: 58,
    color: 'white',
  },
  {
    id: 59,
    color: 'black',
  },
  {
    id: 60,
    color: 'white',
  },
  {
    id: 61,
    color: 'black',
  },
  {
    id: 62,
    color: 'white',
  },
  {
    id: 63,
    color: 'black',
  },
  {
    id: 64,
    color: 'white',
  },
];
