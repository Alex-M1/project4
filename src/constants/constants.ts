export const MESSAGES = {
  invalid_login: 'invalid_login',
  invalid_password: 'invalid_password',
  invalid_confirm: 'invalid_confirm',
};

export const REG_EXP = {
  login: /^[A-Za-z0-9]{1,25}/,
  password: /^[A-Za-z0-9]{6,25}$/,
};

export const COOKIE_SETTING = {
  AGE: 20000,
  SPLIT_LEN: 2,
};

export const TIC_TAC = {
  WINNER_LINES: [
    [0, 1, 2],
    [3, 4, 5],
    [6, 7, 8],
    [0, 3, 6],
    [1, 4, 7],
    [2, 5, 8],
    [0, 4, 8],
    [2, 4, 6],
  ],
  FIELD_SIZE: 480,
};
export const COOKIE_AGE = 20000;

export const GAME_TYPE = {
  TIC_TAC_TOE: 'Tic-tac-toe',
  CHECKERS: 'Checkers',
};

export const TIC_TAC_ITEM = {
  X: 'X',
  O: 'O',
};

export const CHESS_DESK = {
  COLS: [1, 2, 3, 4, 5, 6, 7, 8],
  ROWS: [1, 2, 3, 4, 5, 6, 7, 8],
  CELLS_NUM_INIT: 0,
};

export const LANGUAGE = {
  LANG: 'lang',
  EN: 'en',
  RU: 'ru',
};

export const VIEW_OPTIONS = {
  HEADER_HEIGHT: 90,
  DRAGGIN_OPACITY: {
    IS_DRAGGING: 0.3,
    NOT_DRAGGING: 1,
  },
  CHECKER_FIELD: 500,
  CELL_BACKGROUND: {
    GRAY: '#ccc',
    WHITE: '#fff',
    POSSIBLE: '#b2ff65',
  },
};

export const LOCAL_STORAGE = {
  login: 'login',
  token: 'token',
  gameOptions: 'gameOptions',
};

export const GAME_SETTINGS = {
  bot: 'Bot',
  user: 'User',
};

export const CHECKER_FIELD_INIT = [
  null,
  {
    checker: { blackChecker: false, queen: false },
    index: 1,
    blackSquare: true,
  },
  {
    checker: null,
    index: 2,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 3,
    blackSquare: true,
  },
  {
    checker: null, index: 4, blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 5,
    blackSquare: true,
  },
  {
    checker: null,
    index: 6,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 7,
    blackSquare: true,
  },
  { checker: null, index: 8, blackSquare: false },
  { checker: null, index: 9, blackSquare: false },
  {
    checker: { blackChecker: false, queen: false },
    index: 10,
    blackSquare: true,
  },
  {
    checker: null,
    index: 11,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 12,
    blackSquare: true,
  },
  {
    checker: null,
    index: 13,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 14,
    blackSquare: true,
  },
  {
    checker: null,
    index: 15,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 16,
    blackSquare: true,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 17,
    blackSquare: true,
  },
  {
    checker: null,
    index: 18,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 19,
    blackSquare: true,
  }, {
    checker: null,
    index: 20,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 21,
    blackSquare: true,
  },
  {
    checker: null,
    index: 22,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: false,
      queen: false,
    },
    index: 23,
    blackSquare: true,
  },
  {
    checker: null,
    index: 24,
    blackSquare: false,
  },
  {
    checker: null,
    index: 25,
    blackSquare: false,
  },
  {
    checker: null,
    index: 26,
    blackSquare: true,
  },
  {
    checker: null,
    index: 27,
    blackSquare: false,
  },
  {
    checker: null,
    index: 28,
    blackSquare: true,
  },
  {
    checker: null,
    index: 29,
    blackSquare: false,
  },
  {
    checker: null,
    index: 30,
    blackSquare: true,
  },
  {
    checker: null,
    index: 31,
    blackSquare: false,
  },
  {
    checker: null,
    index: 32,
    blackSquare: true,
  },
  {
    checker: null,
    index: 33,
    blackSquare: true,
  },
  {
    checker: null,
    index: 34,
    blackSquare: false,
  },
  {
    checker: null,
    index: 35,
    blackSquare: true,
  },
  {
    checker: null,
    index: 36,
    blackSquare: false,
  },
  {
    checker: null,
    index: 37,
    blackSquare: true,
  },
  {
    checker: null,
    index: 38,
    blackSquare: false,
  },
  {
    checker: null,
    index: 39,
    blackSquare: true,
  },
  {
    checker: null,
    index: 40,
    blackSquare: false,
  },
  {
    checker: null,
    index: 41,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 42,
    blackSquare: true,
  },
  {
    checker: null,
    index: 43,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 44,
    blackSquare: true,
  },
  {
    checker: null,
    index: 45,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 46,
    blackSquare: true,
  },
  {
    checker: null,
    index: 47,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 48,
    blackSquare: true,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 49,
    blackSquare: true,
  },
  {
    checker: null,
    index: 50,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 51,
    blackSquare: true,
  },
  {
    checker: null,
    index: 52,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 53,
    blackSquare: true,
  },
  {
    checker: null,
    index: 54,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 55,
    blackSquare: true,
  },
  {
    checker: null,
    index: 56,
    blackSquare: false,
  },
  {
    checker: null,
    index: 57,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 58,
    blackSquare: true,
  },
  {
    checker: null,
    index: 59,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 60,
    blackSquare: true,
  },
  {
    checker: null,
    index: 61,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 62,
    blackSquare: true,
  },
  {
    checker: null,
    index: 63,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 64,
    blackSquare: true,
  },
];
