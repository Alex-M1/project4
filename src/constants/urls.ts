export const url = {
  authClient: '/',
  regClient: '/registration',
  server: 'http://35.176.167.155:8089',
  registration: '/registration/reg',
  auth: '/authorization/auth',
};
