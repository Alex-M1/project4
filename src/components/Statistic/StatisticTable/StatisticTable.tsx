import React, { useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { v4 as uuidv4 } from 'uuid';
import { useTheme } from 'src/components/hooks/useTheme';
import { IStatUUID } from 'store/statistic/types';
import StatisticItem from '../StatisticItem';
import { StHeaderWrapper, StResultsWrapper, StWrapper } from './styled';

interface IProps {
  getStatistic: () => void;
  statisticData: Array<IStatUUID>;
}

const StatisticTable: React.FC <IProps> = ({ getStatistic, statisticData }) => {
  useEffect(() => {
    getStatistic();
  }, []);
  const { t } = useTranslation();
  const { colors, theme } = useTheme();
  const statistic = statisticData.map((el) => {
    return <StatisticItem {...el} key={uuidv4()}/>;
  });

  return (
    <StWrapper
          theme={theme}
          colors={colors}
    >
      <StHeaderWrapper
        theme={theme}
        colors={colors}
      >
        <h1>{t('game_statistic')}</h1>
        <div>
          <span>{t('user_name')}</span>
          <span>{t('game_type')}</span>
          <span>{t('opposer')}</span>
          <span>{t('winner')}</span>
        </div>
      </StHeaderWrapper>
      <StResultsWrapper
        theme={theme}
        colors={colors}
      >
        {statistic}
      </StResultsWrapper>
    </StWrapper>
  );
};

export default StatisticTable;
