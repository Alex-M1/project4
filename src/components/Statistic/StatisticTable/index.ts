import { connect } from 'react-redux';
import { AppStateType } from 'store/rootReducer';
import { getStatistic } from 'store/statistic/action';
import { getStatisticByUuid } from 'store/statistic/selectors';
import StatisticTable from './StatisticTable';

const mapStateToProps = (state: AppStateType) => ({
  statisticData: getStatisticByUuid(state),
});
const mapDispatchToProps = {
  getStatistic,
};

export default connect(mapStateToProps, mapDispatchToProps)(StatisticTable);
