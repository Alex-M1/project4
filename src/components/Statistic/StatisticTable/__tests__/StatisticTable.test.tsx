import { GAME_TYPE } from 'constants/constants';
import React from 'react';
import configureStore from 'redux-mock-store';
import { mountSmart, shallowSmart } from 'src/__tests__/testHelper';
import StatisticTable from '../StatisticTable';

jest.mock('react-i18next', () => ({
    useTranslation: jest.fn().mockImplementation(() => ({
        t: jest.fn(),
    })),
}),
);

describe('StatisticTable', () => {
  const mockStore = configureStore();
  const store = mockStore({
    room: {
      rooms: [],
      gameType: GAME_TYPE.CHECKERS,
      toRoom: '',
    },
    ticTac: {
      squares: Array(9).fill(null),
      steps: {
        count: 0,
        isUserStep: null,
      },
      isGameEnd: false,
      isMyTurn: true,
      myOpponentGame: {
        id: null,
        gameType: null,
        guestLogin: null,
        startTime: null,
        stepDtoList: null,
      },
      winner: '',
    },
    statistic: {
      statistic: [
        {
          creatorLogin: 'string',
          gameType: 'string',
          guestLogin: 'string',
          winnerLogin: 'string',
        },
      ],
    },
  });
    describe('StatisticTable', () => {
      let props;
      beforeEach(() => {
        props = {
          getStatistic: jest.fn(),
          statisticData: [
            { 
            creatorLogin: 'string',
            gameType: 'string',
            guestLogin: 'string',
            winnerLogin: 'string',
           }, 
           { 
            creatorLogin: 'string',
            gameType: 'string',
            guestLogin: 'string',
            winnerLogin: 'string',
          }],
        };
      });
      it('Should match snapshot', () => {
        const component = shallowSmart(<StatisticTable {...props} />, store);
        expect(component.html()).toMatchSnapshot();
      });
      it('should render StWrapper', () => {
        const component = mountSmart(<StatisticTable {...props} />, store);
        expect(component.find('styled__StWrapper')).toHaveLength(1);
      });
      it('should render StHeaderWrapper', () => {
        const component = mountSmart(<StatisticTable {...props} />, store);
        expect(component.find('styled__StHeaderWrapper')).toHaveLength(1);
      });
      it('should render StResultsWrapper', () => {
        const component = mountSmart(<StatisticTable {...props} />, store);
        expect(component.find('styled__StResultsWrapper')).toHaveLength(1);
      });
      it('should render h1', () => {
        const component = mountSmart(<StatisticTable {...props} />, store);
        expect(component.find('h1')).toHaveLength(1);
      });
      it('should render div', () => {
        const component = mountSmart(<StatisticTable {...props} />, store);
        expect(component.find('div')).toHaveLength(6);
      });
      it('should render span', () => {
        const component = mountSmart(<StatisticTable {...props} />, store);
        expect(component.find('span')).toHaveLength(12);
      });
  });
});
