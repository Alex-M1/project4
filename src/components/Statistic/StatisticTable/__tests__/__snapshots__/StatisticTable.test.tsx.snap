// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticTable StatisticTable Should match snapshot 1`] = `"<div class=\\"styled__StWrapper-kHlHLr fYkXuW\\"><div class=\\"styled__StHeaderWrapper-bKzUbg fqprgP\\"><h1></h1><div><span></span><span></span><span></span><span></span></div></div><div class=\\"styled__StResultsWrapper-eOGtIh gxuytl\\"><div><span></span><span></span><span>string</span><span>string</span></div><div><span></span><span></span><span>string</span><span>string</span></div></div></div>"`;
